### PR TITLE
Extended load balancer functionality to support persistence profiles,…

### PR DESCRIPTION
**Pull request by tonybaloney**
… irules and default monitors.

### CODE REVIEW

File: libcloud/common/dimensiondata.py [hunk 1]
```diff
@@ -553,3 +553,64 @@ def __repr__(self):
...
+        self.name = name
+        self.compatible_listeners = compatible_listeners
+        self.fallback_compatible = fallback_compatible
+
+    def __repr__(self):
+        return (('<DimensionDataPersistenceProfile: id=%s, name=%s>')
+                % (self.id, self.name))
+
+
+class DimensionDataDefaultiRule(object):
```
Comment (Kami) [inline libcloud/common/dimensiondata.py:706]: Is the `i` in `Defaulti` intentional?

File: libcloud/common/dimensiondata.py [hunk 2]
```diff
@@ -553,3 +553,64 @@ def __repr__(self):
...
+    """
+    A default iRule for a network domain, can be applied to a listener
```
Comment (Kami) [inline libcloud/common/dimensiondata.py:708]: Ah, never mind, I assume i refers to `ingress`?

Comment (Kami) [inline libcloud/common/dimensiondata.py:708]: If so, I would prefer to explicitly call it out - so `DimensionDataDefaultIngressRule`.

File: libcloud/common/dimensiondata.py [hunk 3]
```diff
@@ -553,3 +553,64 @@ def __repr__(self):
...
+    """
+    def __init__(self, id, name, compatible_listeners):
```
Comment (Kami) [inline libcloud/common/dimensiondata.py:710]: Some docstrings for those arguments would be nice - I have no idea what `compatible_listeners` represents (I assume it's an array of something).

File: libcloud/common/dimensiondata.py [hunk 4]
```diff
@@ -553,3 +553,64 @@ def __repr__(self):
...
+    def __repr__(self):
+        return (('<DimensionDataDefaultiRule: id=%s, name=%s>')
+                % (self.id, self.name))
+
+
+class DimensionDataVirtualListenerCompatibility(object):
+    """
+    A compatibility preference for a persistence profile
+    """
+    def __init__(self, type, protocol):
```
Comment (Kami) [inline libcloud/common/dimensiondata.py:739]: Same here.

File: libcloud/common/dimensiondata.py (hunk 2 above)
Comment (tonybaloney) [inline libcloud/common/dimensiondata.py:708]: iRule is the feature name. Can't remember what it stands for! It's a feature of the F5 

Sent from Outlook

On Sat, Dec 5, 2015 at 8:06 PM -0800, "Tomaz Muraus" notifications@github.com wrote:

In libcloud/common/dimensiondata.py:

> -    Fallback Persistence Profile.
> -    """
> -    def **init**(self, id, name, compatible_listeners, fallback_compatible):
> -        self.id = id
> -        self.name = name
> -        self.compatible_listeners = compatible_listeners
> -        self.fallback_compatible = fallback_compatible
>   +
> -    def **repr**(self):
> -        return (('<DimensionDataPersistenceProfile: id=%s, name=%s>')
> -                % (self.id, self.name))
>   +
>   +
>   +class DimensionDataDefaultiRule(object):
> -    """
> -    A default iRule for a network domain, can be applied to a listener

Ah, never mind, I assume i refers to ingress?

**Pull request closed without merging**